### PR TITLE
Gunmods contribute "ammo_effects" to the main weapon

### PR DESCRIPTION
**Pull request by John-Candlebury**
Let me preface this by saying that I have absolutely now Idea of how  this works, I just did random stuff until it worked.

Well with that out of the way, this should implement my ideas from #16296.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
One 'm' in "Emitter"

File: data/json/items/gunmod/laser_gunmods.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    "material": [ "steel" ],
+    "symbol": ":",
+    "color": "light_gray",
+    "location": "emmitter",
+    "mod_targets": [ "pistol", "rifle" ],
+    "min_skills": [ [ "weapon", 4 ], [ "electronics", 5 ] ],
+    "damage_modifier": 10,
+    "range_modifier": 5,    
+    "ups_charges": 20,
+    "ammo_effects" : [ "FLARE" ]
```
Comment (Coolthulhu) [inline data/json/items/gunmod/laser_gunmods.json]: "FLARE" flag starts fires on fire-resistant materials. It implies adding fuel, not just setting things alight.

Comment (DangerNoodle) [inline data/json/items/gunmod/laser_gunmods.json]: Flag that might be best would be `INCENDIARY` if I recall?

Comment (John-Candlebury) [inline data/json/items/gunmod/laser_gunmods.json]: The effect was on purpose, but yes it doesnt make any sense from a realism point of view. I will remove it

Laser weapons already have incendiary, just going to leave the mod at extra damage

**Comment by Coolthulhu** [comment]
One potential problem:
There are gunmods that are guns. Those have their own `ammo_effects` that should NOT be carried to the main gun, but will most likely be in the current code.
Could be simply fixed by renaming the field, to something like `added_ammo_effects`, though that would break the "convention" (ammo and gun both use `ammo_effects`).

This bug would also affect a gun that is ammo, but I don't think anyone cares about guns that are ammo, while gunmods that are guns do exist.

**Comment by John-Candlebury** [comment]
>There are gunmods that are guns. Those have their own ammo_effects that should NOT be carried to the main gun, but will most likely be in the current code.
Could be simply fixed by renaming the field, to something like added_ammo_effects, though that would break the "convention" (ammo and gun both use ammo_effects).

I tested to see if the weapon gunmods transferred their effects to the main weapon and it doesnt seem to happen. I used:
- A combination gun modified so that the underbarrel had the LASER ammo effect and the main weapon didnt happen
- A v29 with electrolaser comversion and a m203 loaded with tear gas shells. The v29 shot the laser with only the beanbag effect, as expected. The m203 also shot the normal shell, without the beanbag effect.

So it appears to work correctly, I dont know how to test for guns that are ammo though 

Also I have style errors but the linter that comes with the repository is not finding the need for any changes, and the web based tool gets errors due to the missing context. What do I do?

**Comment by Coolthulhu** [comment]
Well, if it works, then it's fine.
No need to test ammo-guns - if there are any problems with them, they are also problems in master branch.

**Pull request merged**